src/components/GameItem.js

```diff
@@ -3,27 +3,54 @@ import '../App.css';
 import ItemIcons from '../ItemIcons.js';
 import PropTypes from 'prop-types';
 
+
 class GameItem extends Component {
   static propTypes = {
     height: PropTypes.number.isRequired,
     layer: PropTypes.number.isRequired,
+    type: PropTypes.string.isRequired,
   }
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      spotted: false,
+    };
+  }
+
+  
+
   onItemClicked = () => {
-    // Fill this in for Wave 2!
+    if (this.state.spotted != true ) {
+      this.setState({ spotted: true });
+      this.props.onItemClicked(this.props.type)
```

Comment:
It can be helpful to name this one onItemClickedCallback to make it easier to differentiate the two functions.